spec/classes/chrony_spec.rb

```diff
@@ -26,20 +26,27 @@
         case facts[:osfamily]
         when 'Archlinux'
           context 'using defaults' do
-            it { is_expected.to contain_file('/etc/chrony.conf').with_content(%r{^\s*cmdallow 127\.0\.0\.1$}) }
-            ['0.pool.ntp.org', '1.pool.ntp.org', '2.pool.ntp.org', '3.pool.ntp.org'].each do |s|
-              it { is_expected.to contain_file('/etc/chrony.conf').with_content(%r{^\s*server #{s} iburst$}) }
+            it do
+              is_expected.to contain_file('/etc/chrony.conf')
+                .with_content(%r{^\s*cmdallow 127\.0\.0\.1$})
+                .with_content(%r{^\s*server 0.pool.ntp.org iburst$})
+                .with_content(%r{^\s*server 1.pool.ntp.org iburst$})
+                .with_content(%r{^\s*server 2.pool.ntp.org iburst$})
+                .with_content(%r{^\s*server 3.pool.ntp.org iburst$})
+                .with_content(%r{^\s*rtconutc$})
+                .with_content(%r{^\s*driftfile /var/lib/chrony/drift$})
+                .with_content(%r{^\s*rtcsync$})
+                .with_content(%r{^\s*dumpdir /var/lib/chrony$})
+                .without_content(%r{^\s*\n\s*$})
+            end
+            it do
+              is_expected.to contain_file('/etc/chrony.keys')
```

Comment:
Something I always wonder about is whether you want to copy these to be explicit or have some variable in rspec with this location and reuse the checks for all operatingsystems. Then if you need OS specific checks, only list those. So roughly:
```ruby
let(:keys_file) do
  case facts[:osfamily]
  when 'ArchLinux'
    /etc/chrony.keys
  # ...
  end
end
```

Then later on `it { is_expected.to contain_file(keys_file).with_... }`

That's the approach I take most often because it's less work code but is it a good practice?

Reply:
I'm not qualified to answer that. But I like it.
I also believe that the separate blocks for ```context 'with some params passed in' do``` now could be replaced with a single, since we now don't have separate template files.